packages/core/src/lib/documentPersistedInputs.ts

```diff
@@ -67,7 +67,7 @@ export type PlaygroundInputs<S extends InputSource> = {
   }
   // DEPRECATED: Remove after a while
   dataset: LinkedDataset & { datasetId: number | undefined }
-  datasetV2: Record<number, LinkedDatasetRow>
+  datasetV2: Record<number, LinkedDatasetRow> | undefined
```

Comment:
The fix for the bug on local storage is not to lie here and assume users can have `undefined` this. After changing this I fixed all the places where this key was used.